### PR TITLE
Python implementation for a proposed Go Op. 

### PR DESCRIPTION
**Pull request by abhinavarora**
This PR also contains a `notest_concuurency.py` that is a test that we need to pass when `go op` and channels are implemented in C++

### CODE REVIEW

File: python/paddle/v2/fluid/__init__.py [hunk 1]
```diff
@@ -34,13 +34,14 @@
 from core import LoDTensor, CPUPlace, CUDAPlace
 from distribute_transpiler import DistributeTranspiler
 from distribute_transpiler_simple import SimpleDistributeTranspiler
+from concurrency import *
```
Comment (wangkuiyi) [inline python/paddle/v2/fluid/__init__.py]: According to the [PEP8 style](https://www.python.org/dev/peps/pep-0008/#id23):

> Wildcard imports (from <module> import *) should be avoided, as they make it unclear which names are present in the namespace, confusing both readers and many automated tools.

File: python/paddle/v2/fluid/concurrency.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+__all__ = [
+    'Go',
+    'make_channel',
+    'channel_send',
+    'channel_recv',
+    'channel_close',
+]
+
+
+class Go(BlockGuard):
```
Comment (wangkuiyi) [inline python/paddle/v2/fluid/concurrency.py]: I don't get it why would we need to define Go as a class?

In https://github.com/PaddlePaddle/Paddle/tree/develop/python/paddle/v2/fluid, I don't see a While class or an IfThenElse class.

Comment (kavyasrinet) [reply python/paddle/v2/fluid/concurrency.py]: Oh the `While` (https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/layers/control_flow.py#L604) and `IfThenElse`(https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/layers/control_flow.py#L1243) classes are in https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/layers/control_flow.py

File: python/paddle/v2/fluid/__init__.py (hunk 1 above)
Comment (abhinavarora) [reply python/paddle/v2/fluid/__init__.py]: Thank you for pointing that out. Fixed in the latest commit.

**Pull request merged**